### PR TITLE
add snackbar feature

### PR DESCRIPTION
**Pull request by ydomenjoud**

### CODE REVIEW

**Review by jsp-up — DISMISSED** [review]

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.component.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    private snackbarService: KalSnackbarService) {
+  }
+
+  actionClicked() {
+    this.config.action.callback();
+  }
+
+  ngOnInit() {
+
+    const duration = Math.max(this.config.duration, 1) * 1000;
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.component.ts:43]: For duration that needs to be greater than 1, you could assign its value directly in constructor of the KalSnackbarConfig

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.module.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+export * from './kal-snackbar.component';
+export * from './kal-snackbar.service';
+
+const exports = [
+  KalSnackbarComponent
+];
+
+@NgModule({
+  declarations: exports,
+  exports: [...exports, OverlayModule, PortalModule],
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.module.ts]: Format code

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.service.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import { KalOverlayManager } from '../../utils/classes/kal-overlay-manager';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class KalSnackbarService extends KalOverlayManager {
+
+  private snackbarsList: ComponentRef<KalSnackbarComponent>[] = [];
+
+  private waitingSnackbarsList: KalSnackbarConfig[] = [];
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.service.ts:24]: Comment this

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.service.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import { KalSnackbarComponent } from './kal-snackbar.component';
+import { KAL_SNACKBAR_CONFIG } from './kal-snackbar.injector';
+import { KalOverlayManager } from '../../utils/classes/kal-overlay-manager';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class KalSnackbarService extends KalOverlayManager {
+
+  private snackbarsList: ComponentRef<KalSnackbarComponent>[] = [];
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.service.ts]: Comment this

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.service.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  private waitingSnackbarsList: KalSnackbarConfig[] = [];
+
+  constructor(protected overlay: Overlay,
+              private injector: Injector,
+              @Optional() private location?: Location) {
+    super(overlay);
+  }
+
+  /**
+   * open Dialog
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.service.ts]: Open snackbar

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.service.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      config.overlayRef = overlayRef;
+      config.active = true;
+
+      const snackbarRef = this.attachSnackbar<D>(overlayRef, config);
+
+      // // register dialog
+      this.snackbarsList.push(snackbarRef);
+
+      return snackbarRef;
+    } else {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.service.ts:48]: remove this

**Review by MichaelGrosio — CHANGES_REQUESTED** [review]

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.component.ts (hunk 1 above)
Comment (ydomenjoud) [reply projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.component.ts:43]: it's already a feature provided by this PR

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.module.ts (hunk 2 above)
Comment (ydomenjoud) [reply projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.module.ts]: what do you mean by "format code" ?

Comment (jsp-up) [reply projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.module.ts]: `    exports: [
      ...exports,
      OverlayModule,
      PortalModule
    ];`

Comment (jsp-up) [reply projects/kalidea/kaligraphi/src/lib/atoms/kal-snackbar/kal-snackbar.module.ts]: you have to break lines between variables in exports array

**Review by MichaelGrosio — DISMISSED** [review]

**Review by Bellico — DISMISSED** [review]

**Review by jsp-up — APPROVED** [review]

**Review by MichaelGrosio — APPROVED** [review]

**Pull request merged**